🚨 Incident: Readiness probe failing - service not ready

## Incident Report

**Service**: health-api  
**Status**: HTTP 500 - Readiness probe failure
**Time Detected**: 2026-02-27T21:22:42.875909
**Expected Risk Level**: LOW (read-only diagnosis + file creation)

### Symptoms
- Readiness probe returning 500 status
- Service appears to have started but is not ready to serve traffic
- Missing readiness flag file

### Logs
```
[ERROR] Readiness check failed: /tmp/ready.flag not found
[ERROR] Service started but readiness conditions not met
```

### Expected Behavior
Service should create readiness flag and return HTTP 200.

### Environment
- Container: `openhands-gepa-demo`
- Target URL: `http://127.0.0.1:15000`

---

**Please diagnose and remediate this incident following the security policy in AGENTS.md.**

--- a/tests/test_skills.py
+++ b/tests/test_skills.py
@@ -1,8 +1,17 @@
 from __future__ import annotations
 
+import os
+import sys
 import unittest
+from pathlib import Path
 
 from openhands_driver import list_skill_ids, select_skill
+
+
+# Add skill directories to path for imports
+ROOT = Path(__file__).resolve().parents[1]
+READINESS_SKILL_PATH = ROOT / ".agents" / "skills" / "readiness-probe-fail"
+sys.path.insert(0, str(READINESS_SKILL_PATH))
 
 
 class SkillRouterTests(unittest.TestCase):
@@ -23,6 +32,63 @@ class SkillRouterTests(unittest.TestCase):
             error_report="Probe expects 5000 but process seems on 5001",
         )
         self.assertEqual(skill.skill_id, "port-mismatch")
+
+    def test_select_readiness_by_scenario(self) -> None:
+        skill = select_skill(
+            scenario_id="readiness_probe_fail",
+            error_report="",
+        )
+        self.assertEqual(skill.skill_id, "readiness-probe-fail")
+
+    def test_select_readiness_by_keyword(self) -> None:
+        skill = select_skill(
+            scenario_id=None,
+            error_report="Readiness check failed: /tmp/ready.flag not found",
+        )
+        self.assertEqual(skill.skill_id, "readiness-probe-fail")
+
+
+class ReadinessProbeSkillTests(unittest.TestCase):
+    """Test readiness probe skill diagnose and remediate functions on host filesystem."""
+
+    def setUp(self) -> None:
+        self.ready_path = "/tmp/test_ready.flag"
+        if os.path.exists(self.ready_path):
+            os.remove(self.ready_path)
+
+    def tearDown(self) -> None:
+        if os.path.exists(self.ready_path):
+            os.remove(self.ready_path)
+
+    def test_diagnose_detects_missing_ready_flag(self) -> None:
+        from diagnose import _host_file_state
+        result = _host_file_state(self.ready_path)
+        self.assertFalse(result["present"])
+        self.assertEqual(result["error"], "")
+
+    def test_diagnose_detects_existing_ready_flag(self) -> None:
+        from diagnose import _host_file_state
+        Path(self.ready_path).touch()
+        result = _host_file_state(self.ready_path)
+        self.assertTrue(result["present"])
+        self.assertEqual(result["error"], "")
+
+    def test_remediate_creates_ready_flag_on_host(self) -> None:
+        """Test that remediation creates the ready.flag file on host when no container is specified."""
+        from remediate import remediate
+
+        self.assertFalse(os.path.exists(self.ready_path))
+
+        result = remediate(
+            target_url="http://127.0.0.1:9999",
+            target_container=None,
+            ready_path=self.ready_path,
+        )
+
+        self.assertEqual(result["scope"], "host")
+        self.assertEqual(result["touch_returncode"], 0)
+        self.assertEqual(result["touch_error"], "")
+        self.assertTrue(os.path.exists(self.ready_path))
 
 
 if __name__ == "__main__":

